appstudio-controller/controllers/appstudio.redhat.com/environment_controller.go

```diff
@@ -83,12 +83,24 @@ func (r *EnvironmentReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 			Namespace: req.Namespace,
 		},
 	}
+	gitOpsDeplManagedEnv := managedgitopsv1alpha1.GitOpsDeploymentManagedEnvironment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "managed-environment-" + environment.Name,
+			Namespace: environment.Namespace,
+		},
+	}
 	if err := rClient.Get(ctx, client.ObjectKeyFromObject(environment), environment); err != nil {
-
 		if apierr.IsNotFound(err) {
 			log.Info("Environment resource no longer exists")
 			// A) The Environment resource could not be found: the owner reference on the GitOpsDeploymentManagedEnvironment
 			// should ensure that it is cleaned up, so no more work is required.
+			// As the environment resource no longer exists, the corresponding GitOpsDeploymentManagedEnvironment should be deleted.
+			if err = rClient.Get(ctx, client.ObjectKeyFromObject(&gitOpsDeplManagedEnv), &gitOpsDeplManagedEnv); err == nil {
+				if err = rClient.Delete(ctx, &gitOpsDeplManagedEnv); err != nil {
+					log.Info("Unable to delete GitOpsDeploymentManagedEnvironment")
```

Comment:
Just something minor, but it LGTM. 

Do you think it will be useful to mention the name of the environment and/or the GitOpsDeploymentMangedEnvironment that we are trying to delete?

Reply:
I think it will be useful if something fails and we can know why it is failing like for eg, `Unable to delete GitOpsDeploymentManagedEnvironment`